### DISCUSSION TITLE
Allow port selection to match server plugin settings

### DISCUSSION THREAD

**Discussion by SparkInChief** [coddingtonbear/obsidian-web#71] [Ideas]
While Local REST API for Obsidian [allows](https://github.com/coddingtonbear/obsidian-local-rest-api/blob/64f04f4f4efed9f872e5816304381decc60c1c0d/src/main.ts#L121) changing the listening ports, Obsidian Web hard-codes request ports. 

I am trying out a use case where I needed to change the listener port on the Obsidian plugin for a specific vault, but I discovered I can't change the Chrome extension to match.

https://github.com/coddingtonbear/obsidian-web/blob/ec6de435a5ba80984ffd7ac997036a5043da9e84/src/utils.ts#L160-L161

**Comment by coddingtonbear** [comment]
Unfortunately, I'm not likely to accept that particular feature due to how Chrome Extension security works.  As it stands now, you have to declare patterns for host:port combinations that your plugin will have access to, and the declared host:port combinations are expected to be as tight as possible to make it through the Chrome App Store's review process. You can see this plugin's declarations here: https://github.com/coddingtonbear/obsidian-web/blob/master/public/manifest.json#L38. 

It _might_ be possible to broaden them, but I'm sorry to say that I'd rather not add additional complications to the review process or manage the technical complications it might introduce at this point in time.

**Reply by SparkInChief** [reply]
That totally makes sense and matches what I was realizing when I started working on a PR ... **thank you** for taking the time to have a look even though I should have done more digging before creating the issue.

Workaround: Unpacked version of the awesome plugin that has alternate ports in the manifest, added in Developer Mode for me only.

**Reply by shanness** [reply]
Hey @SparkInChief , just wondering if you have a repo for this?  Just want to avoid unnecessary double ups of whatever work you had to do to make this work?  Can't see you in the forks?

**Reply by asford** [reply]
@coddingtonbear would you be willing to take a minor update to the plugin to expand the manifest's port ranges with 1-2 additional alternative ports? 

E.g. I use separate "professional" and "personal" vaults, configured with different browser profiles. I *suspect* this is one of the more common usage patterns. Declaring just a few port options would go a long way to unblock many multi-vault users.

**Reply by coddingtonbear** [reply]
Unfortunately, just updating the manifest ports wouldn't be enough -- Obsidian Web versions before 3.4 were hard-coded to connect either via HTTP on port 27123 or via HTTPS on 27124; there wasn't a way of passing through a port selection, even if the manifest allowed it.

Good news, though -- as of v  3.4, the extension allows you to connect to Obsidian Local REST API instances running on any port.  Good luck!
